ui/src/app/machines/views/AddMachine/AddMachineForm/AddMachineForm.js

```diff
@@ -46,7 +46,10 @@ const generateMachineSchema = (parametersSchema) =>
     power_type: Yup.string().required("Power type required"),
     pxe_mac: Yup.string()
       .matches(/^([0-9A-Fa-f]{2}:){5}([0-9A-Fa-f]{2})$/, "Invalid MAC address")
-      .required("At least one MAC address required"),
+      .when("power_type", {
+        is: (power_type) => power_type !== "ipmi",
+        then: Yup.string().required("At least one MAC address required"),
+      }),
```

Comment:
hmm, this will test that it's required for everything but IPMI but doesn't test that it's not required for IPMI?

Reply:
The `then` applies the required and is only triggered if `is` is true. Therefore, the logic is if not "ipmi" set required.

Reply:
Please add a test for this